ios/brave-ios/Sources/BraveShields/ShieldStrings.swift

```diff
@@ -345,15 +345,28 @@ extension Strings.Shields {
     comment: "A title for a confirmation window that appears when a user clicks on 'Shred Data'"
   )
 
-  /// An option setting for shredding only when the app is closed
-  public static let shredThisSitesDataConfirmationMessage = NSLocalizedString(
+  /// A message for a confirmation window that appears when a user clicks on 'Shred Data'.
+  /// The placeholder should be replaced with a website domain.
+  private static let shredThisSitesDataConfirmationMessage = NSLocalizedString(
     "ShredThisSitesDataConfirmationMessage",
     tableName: "BraveShared",
     bundle: .module,
-    value: "Are you sure you want to delete this website's data? This action cannot be undone.",
-    comment: "A message for a confirmation window that appears when a user clicks on 'Shred Data'"
+    value: "Are you sure you want to delete '%@' data? This action cannot be undone.",
+    comment: """
+      A message for a confirmation window that appears when a user clicks on 'Shred Data'. 
+      The placeholder will be replaced with a website domain.
+      """
   )
 
+  /// A message for a confirmation window that appears when a user clicks on 'Shred Data'.
+  /// The provided URL will be used to extract the domain or host.
+  public static func shredThisSitesDataConfirmationMessage(for url: URL) -> String {
+    return String(
+      format: Strings.Shields.shredThisSitesDataConfirmationMessage,
+      url.baseDomain ?? url.host ?? url.absoluteString
```

Comment:
IMO this does not belong here and should be put to View classes or an extension

Reply:
What about `Strings+ShieldsPanelView`?

Reply:
to me this logic can be duplicated across two files as I see it's used in two places. It is a very basic logic to use